### PR TITLE
Added Extra Utils and Forestry support back

### PR DESCRIPTION
**Pull request by BloodWorkXGaming**

### CODE REVIEW

File: src/main/java/com/blamejared/compat/extrautils2/Crusher.java [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+        @Override
+        public void apply() {
+            List<IMachineRecipe> toRemove = new ArrayList<>();
+
+            for (IMachineRecipe recipe : XUMachineCrusher.INSTANCE.recipes_registry) {
+                for (Pair<Map<MachineSlotItem, List<ItemStack>>, Map<MachineSlotFluid, List<FluidStack>>> mapMapPair : recipe.getJEIInputItemExamples()) {
+                    for (ItemStack stack : mapMapPair.getKey().get(XUMachineCrusher.INPUT)) {
+                        if (input != null && input.isItemEqual(stack)) {
+                            toRemove.add(recipe);
```
Comment (jaredlll08) [inline src/main/java/com/blamejared/compat/extrautils2/Crusher.java:83]: There is no point in storing the recipe, just remove the recipes here

File: src/main/java/com/blamejared/compat/forestry/BaseAddForestry.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package com.blamejared.compat.forestry;
```
Comment (jaredlll08) [inline src/main/java/com/blamejared/compat/forestry/BaseAddForestry.java]: This stuff should be in MTLib.

File: src/main/java/com/blamejared/compat/forestry/BaseRemoveForestry.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
+package com.blamejared.compat.forestry;
```
Comment (jaredlll08) [inline src/main/java/com/blamejared/compat/forestry/BaseRemoveForestry.java]: This stuff should be in MTLib.

File: src/main/java/com/blamejared/compat/forestry/Carpenter.java [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+	 * Adds shapeless recipe to Carpenter
+	 *
+	 * @param output           recipe product
+	 * @param ingredients      required ingredients
+	 * @param packagingTime    amount of ticks per crafting operation
+	 ** @param fluidInput      required mB of fluid (optional)
+	 ** @param box             required box in top slot (optional)
+	 ** @param remainingItems  no idea (optional)
+	 */
+	//TODO
```
Comment (jaredlll08) [inline src/main/java/com/blamejared/compat/forestry/Carpenter.java]: I would prefer to accept this when all the code is in.

File: src/main/java/com/blamejared/compat/forestry/Still.java [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+	/**
+	 * Adds recipe to Still
+	 *
+	 * @param fluidOutput recipe fluid amount
+	 * @param fluidInput  recipe fluid input
+	 * @param timePerUnit time per crafting operation
+	 */
+	@ZenMethod
+	public static void addRecipe(ILiquidStack fluidOutput, ILiquidStack fluidInput, int timePerUnit) {
+		fluidOutput.amount(fluidOutput.getAmount() / 100);
```
Comment (jaredlll08) [inline src/main/java/com/blamejared/compat/forestry/Still.java]: Magic numbers?

File: src/main/java/com/blamejared/compat/forestry/ThermionicFabricator.java [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+     * @param fluidOutput  recipe fluid amount
+     * @param itemInput    recipe input input
+     * @param meltingPoint point at where itemInput melts down
+     */
+    @ZenMethod
+    public static void addSmelting(int fluidOutput, IItemStack itemInput, int meltingPoint) {
+        //The machines internal tank accept only liquid glass, therefor this function only accept the amount and hardcode the fluid to glass
+        FluidStack fluid = FluidRegistry.getFluidStack("glass", fluidOutput);
+        if (fluid == null){
+            LogHelper.logWarning("Liquid Class is null for the Thermionic Fabricator");
```
Comment (jaredlll08) [inline src/main/java/com/blamejared/compat/forestry/ThermionicFabricator.java]: I think you mean "Liquid Glass"

File: src/main/java/com/blamejared/compat/forestry/ThermionicFabricator.java [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+     * @param ingredients list of input items
+     * @param fluidInput  recipe fluid input
+     *                    * @param plan            recipe plan item (optional)
+     *                    * @param remainingItems  no idea(optional)
+     */
+    @ZenMethod
+    public static void addCast(IItemStack output, IIngredient[][] ingredients, int fluidInput, @Optional IItemStack plan) {
+        FluidStack fluid = FluidRegistry.getFluidStack("glass", fluidInput);
+        if (fluid == null){
+            LogHelper.logWarning("Liquid Class is null for the Thermionic Fabricator");
```
Comment (jaredlll08) [inline src/main/java/com/blamejared/compat/forestry/ThermionicFabricator.java]: I think you mean "Liquid Glass"

File: src/main/java/com/blamejared/util/MTInputHelper.java [hunk 8]
```diff
@@ -0,0 +1,65 @@
+package com.blamejared.util;
```
Comment (jaredlll08) [inline src/main/java/com/blamejared/util/MTInputHelper.java]: This should be in MTLib

**Review by jaredlll08 — CHANGES_REQUESTED** [review]

**Comment by jaredlll08** [comment]
Please don't organize imports. it just creates more clutter for me to read through when reviewing and it just annoying.


Also please add docs here:
https://github.com/jaredlll08/CraftTweaker-Documentation

File: src/main/java/com/blamejared/compat/extrautils2/Crusher.java (hunk 1 above)
Comment (BloodWorkXGaming) [reply src/main/java/com/blamejared/compat/extrautils2/Crusher.java:83]: There indeed is a point, if I try to remove it while I iterate over it I might get a concurrent modification exception.

File: src/main/java/com/blamejared/compat/forestry/BaseAddForestry.java (hunk 2 above)
Comment (BloodWorkXGaming) [reply src/main/java/com/blamejared/compat/forestry/BaseAddForestry.java]: This class is exclusively working for forestry. Why should anyone want to add something to forestry if it is already in modtweaker?

File: src/main/java/com/blamejared/compat/forestry/Still.java (hunk 5 above)
Comment (BloodWorkXGaming) [reply src/main/java/com/blamejared/compat/forestry/Still.java]: It was this way in 1.10 already so I decided to keep it.
They indeed are magic numbers which seem to work.

File: src/main/java/com/blamejared/compat/forestry/Carpenter.java (hunk 4 above)
Comment (BloodWorkXGaming) [reply src/main/java/com/blamejared/compat/forestry/Carpenter.java]: From what I can see it isn't even possible to add shapeless recipes. These are just remainings from the old 1.10 version. Will delete this doc comment + TODO

File: src/main/java/com/blamejared/compat/forestry/Still.java (hunk 5 above)
Comment (BloodWorkXGaming) [reply src/main/java/com/blamejared/compat/forestry/Still.java]: Okey, I tested around a bit and those numbers did exactly nothing. actually, those two lines do absolutely nothing as ILiquidStack is not mutable.
Doing it in the correct actually breaks the recipes, so I removed those two lines.

**Review by jaredlll08 — APPROVED** [review]

**Pull request merged**